pkg/controller/clusters/status.go

```diff
@@ -110,6 +122,70 @@ func updateControlPlaneInitializedCondition(cluster *anywherev1.Cluster, kcp *co
 	conditions.MarkTrue(cluster, anywherev1.ControlPlaneInitializedCondition)
 }
 
+// updateWorkersReadyCondition updates the WorkersReadyConditon condition after checking the state of the worker node groups
+// in the cluster.
+func updateWorkersReadyCondition(cluster *anywherev1.Cluster, machineDeployments []clusterv1.MachineDeployment) {
+	initializedCondition := conditions.Get(cluster, anywherev1.ControlPlaneInitializedCondition)
+	if initializedCondition.Status != "True" {
+		conditions.MarkFalse(cluster, anywherev1.WorkersReadyConditon, initializedCondition.Reason, initializedCondition.Severity, initializedCondition.Message)
```

Comment:
I'm not sure about the reusing the reason here, it might not work well always
What about just using "ControlPlaneNotInitialized" always?
if the user wants know why that is true, they can check the cp condition